robottelo/containers.py

```diff
@@ -0,0 +1,249 @@
+import logging
+
+from uuid import uuid4
+from robottelo.config import settings
+
+logger = logging.getLogger(__name__)
+
+
+class ContainerError(Exception):
+    """Exception raised for failed container management operations."""
+
+
+class DockerContainer:
+    """
+    Gather information needed to spin up a Docker-based container.
+    Attributes:
+            image (init): The name of the container image to build from.
+            tag (init): The tag of the operating system to use.
+            agent (boolian): If set to True, prepares container for
+            katello-agent.
+    """
```

Comment:
can you also mention the example of how to create the docker instances with all options